app/src/main/java/io/intelehealth/client/services/UpdateVisitService.java

```diff
@@ -208,20 +208,51 @@ protected void onHandleIntent(Intent intent) {
                         case ConceptId.TEMPERATURE: {
                             if (obs.getValue() != null && !obs.getValue().trim().isEmpty()) {
                                 try {
-                                    Double fTemp = Double.parseDouble(obs.getValue());
-                                    Double cTemp = ((fTemp - 32) * 5 / 9);
-                                    if (obs.getOpenmrsObsId() != null && !obs.getOpenmrsObsId().equals(0)) {
-
-                                        check = updateObs(UuidDictionary.TEMPERATURE, String.valueOf(cTemp),
-                                                obs.getOpenmrsObsId(), encounterVitals);
-                                        if (!check) check_all = false;
-                                    } else {
-                                        check = createObs(encounterVitals, UuidDictionary.TEMPERATURE, String.valueOf(cTemp),
-                                                String.valueOf(obs.getConceptId()));
-                                        if (!check) check_all = false;
+                                    JSONObject obj = null;
+                                    obj = new JSONObject(String.valueOf(HelperMethods.encodeJSON(this, "vital_config.json")));
+
```

Comment:
@kranthi-mahiti instead of doing this JSONObject call why not just store this in sharedpreferences and call from there?